modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1662,7 +1662,19 @@ void ONNXImporter::parseMul(LayerParams& layerParams, const opencv_onnx::NodePro
         blob = blob.reshape(1, 1);
         if (blob.total() == 1) {
             float blob_value = blob.ptr<float>()[0];
-            float coeff = isDiv ? 1.0 / blob_value : blob_value;
+            float coeff;
+            if ( isDiv )
+            {
+                if ( constId == 0 )
+                {
+                    layerParams.set("power", -1);
+                    coeff = blob_value;
+                }
+                else
+                    coeff = 1.0 / blob_value;
+            }
+            else
+               coeff = blob_value;
```

Comment:
We should always initialize variables to avoid certain cases of bugs.
`Power` layer is calculating `(x*scale + shift)^power`, so, in your case(const/tensor), it will be `1/(x*scale)` instead of `scale/x`. We could use something like:


```cpp
float coeff = blob_value;
if ( isDiv )
{
    coeff = 1.0 / blob_value;
    if ( constId == 0 )
    {
        layerParams.set("power", -1);
    }                    
}
```

But I'm not sure it will be accurate enough.

Reply:
consider input 2/x ,  x/2 ,   I think it should be : 

```

float coeff = blob_value;
if ( isDiv )
{
    
    if ( constId == 0 )
    {
        layerParams.set("power", -1);
    }
  else
        coeff = 1.0 / blob_value;                    
}

```

Reply:
@hujunyao, I just tested your code using the following snippet, which you can use to add tests(it needs input/output saving, you can look at `save_onnx_data_and_model` in `opencv_extra/testdata/dnn/onnx/generate_onnx_models.py`):

```py

node = onnx.helper.make_node(
    'Div',
    inputs=['x', 'y'],
    outputs=['z'],
)

x = np.array([2]).astype(np.float32)
y = np.array([[4, 4], [4, 4]]).astype(np.float32)
z = x / y

X = onnx.helper.make_tensor('x', onnx.TensorProto.FLOAT, x.shape, x)
Y = onnx.helper.make_tensor_value_info('y', onnx.TensorProto.FLOAT, y.shape)
Z = onnx.helper.make_tensor_value_info('z', onnx.TensorProto.FLOAT, z.shape)

graph = onnx.helper.make_graph([node], 'const_div', [Y], [Z], initializer=[X])
model = onnx.helper.make_model(graph, producer_name='const_div')
onnx.save(model, 'const_div.onnx')

```

It outputs a mat of 0.125, which is consistent with my prediction: `1/(2*4)`, whereas the proposed solution outputs 0.5 as expected.

Reply:
I  thought it won't go to this branch when it in 
'''
     if (constId != -1 && haveVariables)
    {
       ..................
    }
    else if (!haveVariables)
'''
  I thought this branch only calculate  two operator : a constant and a variable,  when 2 constants div , it will go to !haveVariable.  but I'm sure that.